src/tacacs_plus_fmt_plug.c

```diff
@@ -0,0 +1,347 @@
+/*
+ * Format for cracking TACACS+ hashes.
+ *
+ * https://insinuator.net/2015/06/tacacs-module-for-loki/
+ *
+ * This software is Copyright (c) 2015, Daniel Mende <dmende [at] ernw.de> and
+ * Copyright (c) 2017, Dhiru Kholia <dhiru [at] openwall.com>, and it is hereby
+ * released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above
+ *    copyright notice, this list of conditions and the following disclaimer
+ *    in the documentation and/or other materials provided with the
+ *    distribution.
+ *
+ *  * Neither the name of the copyright holders nor the names of its
+ *    contributors may be used to endorse or promote products derived from
+ *    this software without specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ *    A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ *    OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *    SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *    LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *    DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *    THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *    OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_tacacsplus;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_tacacsplus);
+#else
+
+#include <string.h>
+
+#ifdef _OPENMP
+#include <omp.h>
+#ifndef OMP_SCALE
+#define OMP_SCALE               32
```

Comment:
Did you tune this or is this arbitrary? I suggested 64 and 32 as a split of 2048, but 2048 itself was probably not tuned, right? My actual guess is that a smaller value might be more optimal here, e.g. 64 and 2 was about right for another format I tuned recently.

Reply:
I haven't tuned this format yet. I also made an assumption that the `2048` value was tuned already. It was probably not a wise decision ;(